proto/common/healthcheck/healthcheck.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package conduit.common.healthcheck;
+
+enum CheckStatus {
+    OK = 0;
+    FAIL = 1;
+    ERROR = 2;
+}
+
+message CheckResult {
+    string SubsystemName = 1;
+    string CheckDescription = 2;
+    CheckStatus Status = 3;
+    string FriendlyMessageToUser = 4;
+}
+
+message Check {
+    repeated CheckResult results = 1;
+    CheckStatus overallStatus = 2;
+}
+
+message SelfCheckRequest {}
+
+message SelfCheckResponse {
+    repeated CheckResult results = 1;
```

Comment:
related to previous comment, is `SelfCheckResponse` redundant with `Check`?

Reply:
I would always use envelopes for both request and response in gRPC. ProtocolBuffers is very particular about how it maintains backwards compatibility and trying to simplify these messages often result in a situation where we need to create a new `rpc` because we can't change a response or request without breaking every client. e.g. [_when_ we need to add a parameter to `ListPods`, say we only want pods from a given namespace, we will have to create another `rpc` or break all clients because we don't have an envelope](https://github.com/runconduit/conduit/blob/e328db7e8740a4d01c99ba8c8bc05e236cf95d72/proto/public/api.proto#L1350).

Reply:
Makes sense to me, and I agree that all rpc definitions should use request / response envelopes. I think this is clearer now that the `Check` message has been removed.